packages/subproviders/test/unit/eth_lightwallet_subprovider_test.ts

```diff
@@ -0,0 +1,185 @@
+import { chaiSetup } from '../chai_setup';
+import { ganacheSubprovider } from '../utils/ganache_subprovider';
+import { reportCallbackErrors } from '../utils/report_callback_errors';
+
+chaiSetup.configure();
+const expect = chai.expect;
+
+const FAKE_ADDRESS = '0x44be42fd88e22387c43ba9b75941aa3e680dae25';
+const NUM_GENERATED_ADDRESSES = 10;
+const PASSWORD = 'supersecretpassword99';
+const SEED_PHRASE = 'dilemma hollow outer pony cube season start stereo surprise when edit blast';
+const SALT = 'kvODghzs7Ff1uqHyI0P3wI4Hso4w4iWT2e9qmrWz0y4';
+const HD_PATH_STRING = `m/44'/60'/0'`;
+
+describe('EthLightwalletSubprovider', () => {
+    let ethLightwalletSubprovider: EthLightwalletSubprovider;
+    before(async () => {
+        const options = {
+            password: PASSWORD,
+            seedPhrase: SEED_PHRASE,
+            salt: SALT,
+            hdPathString: HD_PATH_STRING,
+        };
+
+        const createVaultAsync = async (vaultOptions: lightwallet.VaultOptions) => {
+            return new Promise<lightwallet.keystore>(resolve => {
+                lightwallet.keystore.createVault(vaultOptions, (err: Error, vaultKeystore) => {
+                    if (err) {
+                        throw new Error(`Failed to createVault: ${err}`);
+                    }
+                    resolve(vaultKeystore);
+                });
+            });
+        };
+        const deriveKeyFromPasswordAsync = async (vaultKeystore: lightwallet.keystore) => {
+            return new Promise<Uint8Array>(resolve => {
+                vaultKeystore.keyFromPassword(PASSWORD, (err: Error, passwordDerivedKey: Uint8Array) => {
+                    if (err) {
+                        throw new Error(`Failed to get key from password: ${err}`);
+                    }
+                    resolve(passwordDerivedKey);
+                });
+            });
+        };
+        const keystore: lightwallet.keystore = await createVaultAsync(options);
+        const pwDerivedKey: Uint8Array = await deriveKeyFromPasswordAsync(keystore);
+
+        // Generate 10 addresses
+        keystore.generateNewAddress(pwDerivedKey, NUM_GENERATED_ADDRESSES);
+
+        ethLightwalletSubprovider = new EthLightwalletSubprovider(keystore, pwDerivedKey);
+    });
+    describe('direct method calls', () => {
+        describe('success cases', () => {
+            it('returns a list of accounts', async () => {
+                const accounts = await ethLightwalletSubprovider.getAccountsAsync();
+                expect(accounts[0]).to.be.equal(FAKE_ADDRESS);
+                expect(accounts.length).to.be.equal(NUM_GENERATED_ADDRESSES);
+            });
+            it('signs a personal message hash', async () => {
+                const signingAccount = (await ethLightwalletSubprovider.getAccountsAsync())[0];
+
+                // Keccak-256 hash of 'hello world'
+                const messageHash = '0x47173285a8d7341e5e972fc677286384f802f8ef42a5ec5f03bbfa254cb01fad';
```

Comment:
Does this need to be a hex encoded string? Our other tests just use the string 'hello world' and tests against known fixture data.